sdk/tables/azure-tables/review/tables.api.review.md

```diff
@@ -0,0 +1,695 @@
+  ): Promise<DeleteEntityResponse>;
+  deleteTable(tableName: string, options?: DeleteTableOptions): Promise<DeleteTableResponse>;
+  static fromConnectionString(
+    connectionString: string,
+    options?: TableServiceClientOptions
+  ): TableServiceClient;
+  getAccessPolicy(
+    tableName: string,
+    options?: GetAccessPolicyOptions
+  ): Promise<GetAccessPolicyResponse>;
+  getEntity<T extends object>(
+    tableName: string,
+    partitionKey: string,
+    rowKey: string,
+    options?: GetEntityOptions
+  ): Promise<GetEntityResponse<T>>;
+  getProperties(options?: GetPropertiesOptions): Promise<GetPropertiesResponse>;
+  getStatistics(options?: GetStatisticsOptions): Promise<GetStatisticsResponse>;
+  listEntities<T extends object>(
+    tableName: string,
+    query?: QueryOptions,
+    options?: ListEntitiesOptions
+  ): Promise<ListEntitiesResponse<T>>;
+  listTables(query?: QueryOptions, options?: ListTablesOptions): Promise<ListTablesResponse>;
+  setAccessPolicy(
+    tableName: string,
+    acl?: SignedIdentifier[],
+    options?: SetAccessPolicyOptions
+  ): Promise<SetAccessPolicyResponse>;
+  setProperties(
+    properties: ServiceProperties,
+    options?: SetPropertiesOptions
+  ): Promise<SetPropertiesResponse>;
+  updateEntity<T>(
+    tableName: string,
+    entity: Entity<T>,
+    mode?: UpdateMode,
+    eTag?: string,
+    options?: UpdateEntityOptions
+  ): Promise<UpdateEntityResponse>;
+  upsertEntity<T>(
+    tableName: string,
+    entity: Entity<T>,
+    mode?: UpdateMode,
+    options?: UpsertEntityOptions
+  ): Promise<MergeEntityResponse>;
+}
+
+// @public
+export interface TableServiceClientOptions extends coreHttp.ServiceClientOptions {
+  endpoint?: string;
+  version?: string;
+}
+
+// @public
+export interface TableServiceStats {
+  geoReplication?: GeoReplication;
+}
+
+// @public
+export interface TableSetAccessPolicyHeaders {
+  clientRequestId?: string;
+  date?: Date;
+  requestId?: string;
+  version?: string;
+}
+
+// @public
+export interface TableSetAccessPolicyOptionalParams extends coreHttp.OperationOptions {
+  requestId?: string;
+  tableAcl?: SignedIdentifier[];
+  timeout?: number;
+}
+
+// @public
+export interface TableUpdateEntityHeaders {
+  clientRequestId?: string;
+  date?: Date;
+  eTag?: string;
+  requestId?: string;
+  version?: string;
+}
+
+// @public
+export interface TableUpdateEntityOptionalParams extends coreHttp.OperationOptions {
+  ifMatch?: string;
+  queryOptions?: GeneratedQueryOptions;
+  requestId?: string;
+  tableEntityProperties?: {
+    [propertyName: string]: any;
+  };
+  timeout?: number;
+}
+
+// @public
+export type UpdateEntityOptions = Omit<
+  TableUpdateEntityOptionalParams,
+  "tableEntityProperties" | "ifMatch"
+>;
+
+// @public
+export type UpdateEntityResponse = TableUpdateEntityHeaders & {
+  _response: coreHttp.HttpResponse & {
+    parsedHeaders: TableUpdateEntityHeaders;
+  };
+};
+
+// @public
+export enum UpdateMode {
```

Comment:
In Service Bus we've been eschewing enums in favor of just using string unions - no sure if that's an option you'd like to try here or if the 0/1 is better for clients.

So:

export type UpdateMode = "merge" | "replace";